src/tqec/templates/library/spatial.py

```diff
@@ -0,0 +1,430 @@
+"""Provide functions building the :class:`~tqec.templates.rpng.RPNGTemplate`
+instances representing spatial junctions."""
+
+from typing import Literal
+
+from tqec.compile.specs.enums import JunctionArms
+from tqec.exceptions import TQECException
+from tqec.plaquette.enums import MeasurementBasis, ResetBasis
+from tqec.plaquette.frozendefaultdict import FrozenDefaultDict
+from tqec.plaquette.rpng import RPNGDescription
+from tqec.templates.indices.qubit import (
+    QubitHorizontalBorders,
+    QubitSpatialJunctionTemplate,
+    QubitVerticalBorders,
+)
+from tqec.templates.rpng import RPNGTemplate
+
+
+def get_spatial_junction_qubit_template(
+    external_stabilizers: Literal["x", "z"],
+    arms: JunctionArms,
+    reset: ResetBasis | None = None,
+    measurement: MeasurementBasis | None = None,
+) -> RPNGTemplate:
+    """Implementation of a logical qubit performing a spatial junction.
+
+    This function returns a RPNGTemplate instance representing a logical qubit
+    that is touched by 2 or more spatial junctions. The returned template is
+    carefully crafted to avoid hook errors damaging the logical distance.
```

Comment:
```suggestion
    """Implementation of a logical qubit acting as a spatial junction.

    This function returns a RPNGTemplate instance representing a logical qubit
    connecting to others in space and forming a spatial junction. The returned template is
    carefully crafted to avoid hook errors damaging the logical distance.
```